test/resource/igniter_test.exs

```diff
@@ -111,4 +111,99 @@ defmodule Ash.Resource.IgniterTest do
       |> assert_unchanged()
     end
   end
+
+  describe "add_new_code_interface" do
+    setup do
+      igniter =
+        test_project()
+        |> Map.put(:args, %Igniter.Mix.Task.Args{
+          positional: %{
+            resource: "MyApp.Accounts.User"
+          },
+          options: [
+            domain: "MyApp.Accounts",
+            default_actions: ~w[create read update destroy],
+            uuid_primary_key: "id",
+            attribute: ["name:string:required:public"],
+            relationship: [],
+            extend: []
+          ]
+        })
+        |> Mix.Tasks.Ash.Gen.Resource.igniter()
+        |> apply_igniter!()
+
+      %{igniter: igniter}
+    end
+
+    test "add_new_code_interface adds a new code interface to existing resource", %{
```

Comment:
This is the test case that would fail previously, without the do-block on the resource the define is potentially put on the next do-block in the file.